### PR TITLE
parallel-workload: Add webhook sources and POST to them

### PR DESCRIPTION
**Pull request by def-**
Fixes https://github.com/MaterializeInc/materialize/issues/20673. This is pretty basic (as is everything in parallel-workload). Tell me if any specific features should be stressed in particular.

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/parallel_workload/database.py [hunk 1]
```diff
@@ -175,7 +200,64 @@ def create(self, exe: Executor) -> None:
...
+        # TODO: INCLUDE HEADERS (as map)
+        # TODO: CHECK WITH
+
+    def __str__(self) -> str:
+        if self.rename:
+            return f"wh{self.source_id}_{self.rename}"
+        return f"wh{self.source_id}"
+
+    def create(self, exe: Executor) -> None:
+        query = f"CREATE SOURCE {self} IN CLUSTER {self.cluster} FROM WEBHOOK BODY FORMAT {self.body_format.name}"
```
Comment (philip-stoev) [inline misc/python/materialize/parallel_workload/database.py:254]: can you also do `INCLUDE HEADERS` and then
 ```
 CHECK (
      WITH (
        HEADERS,
        BODY AS request_body) LENGHTH(request_body) > 0 )
```

That is, exercise a few more code paths in the webhook sources while still having all the incoming POSTs pass the `CHECK` check.

Comment (philip-stoev) [reply misc/python/materialize/parallel_workload/database.py:254]: Oh I see you are including some headers already below, so I guess only do an always-true `CHECK` clause then.

**Review by philip-stoev — APPROVED** [review]
Thank you!

File: misc/python/materialize/parallel_workload/action.py [hunk 2]
```diff
@@ -329,10 +334,12 @@ def run(self, exe: Executor) -> None:
                 return
             view_id = self.db.view_id
             self.db.view_id += 1
-            # Only use tables for now since LIMIT 1 and statement_timeout are
+            # Don't use views for now since LIMIT 1 and statement_timeout are
             # not effective yet at preventing long-running queries and OoMs.
-            base_object = self.rng.choice(self.db.tables)
-            base_object2: Table | None = self.rng.choice(self.db.tables)
+            base_object = self.rng.choice(self.db.tables + self.db.sources)
+            base_object2: Table | None = self.rng.choice(
```
Comment (nrainer-materialize) [inline misc/python/materialize/parallel_workload/action.py:340]: So, a source is also of type `Table`?

File: misc/python/materialize/parallel_workload/action.py [hunk 3]
```diff
@@ -133,10 +138,10 @@ def errors_to_ignore(self) -> list[str]:
         return result
 
     def run(self, exe: Executor) -> None:
-        tables_views: list[DBObject] = [*self.db.tables, *self.db.views]
-        table = self.rng.choice(tables_views)
+        objects: list[DBObject] = [*self.db.tables, *self.db.views, *self.db.sources]
+        table = self.rng.choice(objects)
         column = self.rng.choice(table.columns)
-        table2 = self.rng.choice(tables_views)
+        table2 = self.rng.choice(objects)
```
Comment (nrainer-materialize) [inline misc/python/materialize/parallel_workload/action.py:144]: Nit: `object2`

File: misc/python/materialize/parallel_workload/data_type.py [hunk 4]
```diff
@@ -81,7 +81,29 @@ def name() -> str:
...
+
+class Jsonb(DataType):
+    @staticmethod
+    def name() -> str:
+        return "jsonb"
+
+    @staticmethod
+    def value(rng: random.Random, in_query: bool = False) -> Any:
+        result = rng.randint(-100, 100)
+        return f"'{result}'::jsonb"
```
Comment (nrainer-materialize) [inline misc/python/materialize/parallel_workload/data_type.py:95]: Is this a valid JSON string?

File: misc/python/materialize/parallel_workload/database.py [hunk 5]
```diff
@@ -175,7 +200,64 @@ def create(self, exe: Executor) -> None:
...
+                self.body_format.to_data_type(),
+                False,
+                self,
+            )
+        ] + [
+            WebhookColumn(include_header, Text, True, self)
+            for include_header in self.include_headers
+        ]
+        # TODO: INCLUDE HEADERS (as map)
+        # TODO: CHECK WITH
```
Comment (nrainer-materialize) [inline misc/python/materialize/parallel_workload/database.py:246]: Unresolved TODOs? Or are they for a follow-up PR?

**Review by nrainer-materialize — APPROVED** [review]

File: misc/python/materialize/parallel_workload/database.py (hunk 5 above)
Comment (def-) [reply misc/python/materialize/parallel_workload/database.py:246]: For a follow-up. I was documented the parts of webhook sources that are still missing.

File: misc/python/materialize/parallel_workload/data_type.py (hunk 4 above)
Comment (def-) [reply misc/python/materialize/parallel_workload/data_type.py:95]: It's a very simple json string, but yes, numbers are valid json:
```
deen=> select '100'::jsonb;
 jsonb
-------
 100
(1 row)
```

File: misc/python/materialize/parallel_workload/action.py (hunk 2 above)
Comment (def-) [reply misc/python/materialize/parallel_workload/action.py:340]: No, was expecting mypy to detect this.

**Review by ParkMyCar — APPROVED** [review]

**Pull request merged**